src/tasks/fetch.ts

```diff
@@ -1,11 +1,39 @@
+import assign from "../util/assign";
+import prefixKeys from "../util/prefixKeys";
+
 import Task from "./task";
+import { asyncGetEntry, normalizeEntry } from "../lib/resourceTiming";
 
-export default class Fetch extends Task {
+class Fetch extends Task {
   public constructor(config: Config, taskData: TaskData) {
     super(config, taskData);
   }
 
-  protected test(): Promise<any> {
-    return Promise.resolve();
+  private fetchObjectAndId(): Promise<string> {
+    return fetch(this.data.resource).then(
+      (res): string => res.headers.get("X-Datacenter") || ""
+    );
+  }
+
+  protected test(): Promise<ResourceTimingEntry> {
+    let subjectId: string;
+
+    return Promise.all([
+      this.fetchObjectAndId(),
+      asyncGetEntry(this.data.resource)
+    ])
+      .then(
+        ([id, entry]): ResourceTimingEntry => {
+          subjectId = id;
+          return normalizeEntry(entry);
+        }
+      )
+      .then(
+        (timing): ResourceTimingEntry => {
+          return prefixKeys(assign({ id: subjectId }, timing), "subject_");
+        }
+      );
```

Comment:
here again, don't understand why we're doing the double-then with the closure when we could just do:
```js
  protected test(): Promise<ResourceTimingEntry> {
    return Promise.all([
      this.fetchObjectAndId(),
      asyncGetEntry(this.data.resource)
    ])
      .then(
        ([id, entry]): ResourceTimingEntry => {
          const timing = normalizeEntry(entry);
          return prefixKeys(assign({ id }, timing), "subject_");
        }
      );
      .catch(error => ...)
  }
```